lock-api/src/main/java/com/palantir/lock/client/LockRefresher.java

```diff
@@ -69,4 +76,9 @@ public void registerLock(LockToken token) {
     public void unregisterLocks(Collection<LockToken> tokens) {
         tokensToRefresh.removeAll(tokens);
     }
+
+    @Override
+    public void close() throws Exception {
+        executor.shutdown();
```

Comment:
This seems odd as this class doesn't create the executor, so it shouldn't manage its lifecycle.

Reply:
hmmm so should we keep a handle to it in LockRefreshingTimeLockService and just call shutdown there?

Reply:
Yeah, that’s probably best, though really we should be injecting one from the app container